webgoat-lessons/sql-injection/src/main/java/org/owasp/webgoat/sql_injection/introduction/SqlInjectionLesson8.java

```diff
@@ -0,0 +1,142 @@
+                log(connection, query);
+                ResultSet results = statement.executeQuery(query);
+
+                if (results.getStatement() != null) {
+                    if (results.first()) {
+                        output.append(generateTable(results));
+                        results.last();
+
+                        if (results.getRow() > 1) {
+                            // more than one record, the user succeeded
+                            return success(this).feedback("sql-injection.8.success").output(output.toString()).build();
+                        } else {
+                            // only one record
+                            return failed(this).feedback("sql-injection.8.one").output(output.toString()).build();
+                        }
+
+                    } else {
+                        // no results
+                        return failed(this).feedback("sql-injection.8.no.results").build();
+                    }
+                } else {
+                    return failed(this).feedback("sql-injection.error").build();
+                }
+            } catch (SQLException e) {
+                return failed(this).feedback("sql-injection.error").output("<br><span class='feedback-negative'>" + e.getMessage() + "</span>").build();
+            }
+
+        } catch (Exception e) {
+            return failed(this).feedback("sql-injection.error").output("<br><span class='feedback-negative'>" + e.getMessage() + "</span>").build();
+        }
+    }
+
+    public static String generateTable(ResultSet results) throws SQLException {
+        ResultSetMetaData resultsMetaData = results.getMetaData();
+        int numColumns = resultsMetaData.getColumnCount();
+        results.beforeFirst();
+        StringBuffer table = new StringBuffer();
+        table.append("<table>");
+
+        if (results.next()) {
+            table.append("<tr>");
+            for (int i = 1; i < (numColumns + 1); i++) {
+                table.append("<th>" + resultsMetaData.getColumnName(i) + "</th>");
+            }
+            table.append("</tr>");
+
+            results.beforeFirst();
+            while (results.next()) {
+                table.append("<tr>");
+                for (int i = 1; i < (numColumns + 1); i++) {
+                    table.append("<td>" + results.getString(i) + "</td>");
+                }
+                table.append("</tr>");
+            }
+
+        } else {
+            table.append("Query Successful; however no data was returned from this query.");
+        }
+
+        table.append("</table>");
+        return (table.toString());
+    }
+
+    public static void log(Connection connection, String action) {
+        action = action.replace('\'', '"');
+        Calendar cal = Calendar.getInstance();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.